main.js

```diff
@@ -1,31 +1,122 @@
+}
+
+function moveWest() {
+  var i = car.location[0]
+  var nextNum = i - 16
+  var stringNum = nextNum.toString()
+  $car.style.left = stringNum + "px"
+  var newNum = parseInt(stringNum)
+  car.location.splice(0, 1, newNum)
+}
+
+function moveSouth() {
+  var i =car.location[1]
+  var nextNum=i+16
+  var stringNum = nextNum.toString()
+  $car.style.top = stringNum + "px"
+  var newNum=parseInt(stringNum)
+  car.location.splice(1,1,newNum)
+}
+
+function moveNorth() {
+  var i = car.location[1]
+  var nextNum = i - 16
+  var stringNum = nextNum.toString()
+  $car.style.top = stringNum + "px"
+  var newNum = parseInt(stringNum)
+  car.location.splice(1, 1, newNum)
 }
 
 document.addEventListener("keydown",function(event){
+
   if (event.code==="ArrowRight"){
-    $car.className= data.east
-    var start=setInterval(moveEast,16)
-    document.addEventListener("keydown",function(){
-      clearInterval(start)
-    })
+    car=new Car("east")
+    $car.className= "car east transition"
+
+    var x = getComputedStyle($car).left
+    var y = getComputedStyle($car).top
+    car.location.push(parseInt(x))
+    car.location.push(parseInt(y))
+
+    clearInterval(start)
+    car.started = false
+
+    if (car === null ||car.started===false){
+      start=setInterval(moveEast,16)
+      car.started=true
+    }
+
   } else if (event.code==="ArrowDown"){
-    $car.className= data.south
+    car =new Car("south")
+    $car.className= "car south transition"
+
+    var x = getComputedStyle($car).left
+    var y = getComputedStyle($car).top
+    car.location.push(parseInt(x))
+    car.location.push(parseInt(y))
+
+    clearInterval(start)
+    car.started = false
+
+    if (car === null || car.started === false){
+      start=setInterval(moveSouth,16)
+      car.started = true
+    }
+
   } else if (event.code==="ArrowLeft"){
-    $car.className=data.west
+    car =new Car("west")
+    $car.className= "car west transition"
+
+    var x = getComputedStyle($car).left
+    var y = getComputedStyle($car).top
+    car.location.push(parseInt(x))
+    car.location.push(parseInt(y))
+
+    clearInterval(start)
+    car.started = false
+
+    if (car === null || car.started === false){
+      start = setInterval(moveWest,16)
+      car.started = true
+    }
+
   } else if (event.code==="ArrowUp"){
-    $car.className=data.north
+    car =new Car("north")
+    $car.className= "car north transition"
+
+    var x = getComputedStyle($car).left
+    var y = getComputedStyle($car).top
+    car.location.push(parseInt(x))
+    car.location.push(parseInt(y))
+
+    clearInterval(start)
+    car.started=false
+
+    if (car === null || car.started === false){
+      start=setInterval(moveNorth,16)
+      car.started = true
+    }
+
+  }else if (event.code==="Space"){
+    clearInterval(start)
+    car.started=false
   }
 })
-
-var data = {
-  north: "car north transition",
-  south: "car south transition",
-  east: "car east transition",
-  west: "car west transition"
-}
```

Comment:
This is getting overcomplicated, a few things to keep in mind
- Location should not be an array, unless the first and second values are x and y coordinates
- You don't want to destroy your data model when you switch directions, it throws your data off sync
- Have a static data model that you update in your interval function or when the user presses a key
- If you have one interval that relies heavily on your data model you can manipulate the state of your data model and have your interval act differently using logic.  If you do it right you may not even have to clear your interval
- Attempt a refactor and let me know if you have questions

Reply:
I wasn't sure how to have just one interval, but other than that I made the data model static and update whenever the keys are pressed instead of creating a new one every time the keys are pressed.